### PR TITLE
Fix bug: switches were not removed from index when all edges were removed

### PR DESCRIPTION
**Pull request by miovd**
Signed-off-by: VEDELAGO MIORA <miora.ralambotiana@rte-france.com>

**Please check if the PR fulfills these requirements** *(please use `'[x]'` to check the checkboxes, or submit the PR and then click the checkboxes)*
- [x] The commit message follows our guidelines
- [x] Tests for the changes have been added (for bug fixes / features)
- [ ] Docs have been added / updated (for bug fixes / features)

**Does this PR already have an issue describing the problem ?** *If so, link to this issue using `'#XXX'` and skip the rest*
No

**What kind of change does this PR introduce?** *(Bug fix, feature, docs update, ...)*
Bug fix

**What is the current behavior?** *(You can also link to an open issue here)*
Switches are not removed from index and still appear (for example in `getSwitchesCount()`) after all edges removed. 

**Does this PR introduce a breaking change or deprecate an API?** *If yes, check the following:*
Yes
- [x] The *Breaking Change* or *Deprecated* label has been added
- [ ] The migration guide has been updated in the github wiki *(What changes might users need to make in their application due to this PR?)*

### CODE REVIEW

File: iidm/iidm-impl/src/main/java/com/powsybl/iidm/network/impl/BusBreakerVoltageLevel.java [hunk 1]
```diff
@@ -353,7 +353,7 @@ public void edgeRemoved(int e, SwitchImpl obj) {
             }
 
             @Override
-            public void allEdgesRemoved() {
+            public void allEdgesRemoved(Collection<SwitchImpl> obj) {
                 invalidateCache();
```
Comment (flo-dup) [inline iidm/iidm-impl/src/main/java/com/powsybl/iidm/network/impl/BusBreakerVoltageLevel.java:357]: Sorry I'm missing something here, why didn't we remove the switches from the network index in bus breaker view?

File: iidm/iidm-impl/src/main/java/com/powsybl/iidm/network/impl/NodeBreakerVoltageLevel.java [hunk 2]
```diff
@@ -528,18 +528,14 @@ public void edgeRemoved(int e, SwitchImpl aSwitch) {
...
+            public void allEdgesRemoved(Collection<SwitchImpl> aSwitches) {
                 NetworkImpl network = getNetwork();
-                for (SwitchImpl s : graph.getEdgesObject()) {
-                    if (s != null) {
-                        network.getListeners().notifyBeforeRemoval(s);
-                        removedSwitchesIds.add(s.getId());
-                        network.getIndex().remove(s);
-                    }
-                }
+                aSwitches.forEach(ss -> {
```
Comment (flo-dup) [inline iidm/iidm-impl/src/main/java/com/powsybl/iidm/network/impl/NodeBreakerVoltageLevel.java:533]: Is there any reason to have a difference between the `aSwitches` passed on and the `switches` map?

**Review by flo-dup — COMMENTED** [review]
It's worth adding an assert of `Network::getSwitchCount` in `AbstractNodeBreakerInternalConnectionsTest::testRemoveVoltageLevelWithInternalConnectionsIssue`, don't you think? or a specific test

File: iidm/iidm-impl/src/main/java/com/powsybl/iidm/network/impl/BusBreakerVoltageLevel.java (hunk 1 above)
Comment (miovd) [reply iidm/iidm-impl/src/main/java/com/powsybl/iidm/network/impl/BusBreakerVoltageLevel.java:357]: We do remove the switches from the network index in bus breaker view but it is not done in the graph listener (I don't actually remember why)

File: iidm/iidm-impl/src/main/java/com/powsybl/iidm/network/impl/NodeBreakerVoltageLevel.java (hunk 2 above)
Comment (miovd) [reply iidm/iidm-impl/src/main/java/com/powsybl/iidm/network/impl/NodeBreakerVoltageLevel.java:533]: `aSwitches` contains the list of switches while `switches` contains ids and the associated edge's number

**Comment by miovd** [comment]
> `testRemoveVoltageLevelWithInternalConnectionsIssue`

This test is not really related as the issue appeared only with `removeAllSwitches` (when removing a whole voltage level for example). However, I added a test in `DefaultNetworkReducerTest`

**Review by flo-dup — APPROVED** [review]

**Pull request merged**